### PR TITLE
Registration

### PR DESCRIPTION
**Pull request by dbkr**

### CODE REVIEW

File: lib/client.js [hunk 1]
```diff
@@ -1381,6 +1385,47 @@ function _PojoToMatrixEventMapper(plainOldJsObject) {
...
+ * @param {string} email
+ * @param {string} client_secret
+ * @param {string} send_attempt
+ * @param {string} next_link Optional
+ * @param {module:client.callback} callback Optional.
+ * @return {module:client.Promise} Resolves: TODO
+ * @return {module:http-api.MatrixError} Rejects: with an error response.
+ */
+MatrixClient.prototype.requestEmailToken = function(email, client_secret,
+                                                    send_attempt, next_link, callback) {
```
Comment (Kegsay) [inline lib/client.js]: `camelCase` please.

File: lib/http-api.js [hunk 2]
```diff
@@ -352,6 +387,7 @@ var requestCallback = function(defer, userDefinedCallback, onlyData) {
     return function(err, response, body) {
         if (!err && response.statusCode >= 400) {
             err = new module.exports.MatrixError(body);
+            err.httpStatus = response.statusCode;
```
Comment (Kegsay) [inline lib/http-api.js]: Good idea. Please add this property to the JSDoc at the bottom of this file.

File: lib/client.js [hunk 3]
```diff
@@ -807,15 +808,6 @@ MatrixClient.prototype.publicRooms = function(callback) {
  * @return {module:client.Promise} Resolves: TODO
  * @return {module:http-api.MatrixError} Rejects: with an error response.
  */
-MatrixClient.prototype.registerFlows = function(callback) {
```
Comment (Kegsay) [inline lib/client.js]: Is there no alternative for this in v2?

**Comment by Kegsay** [comment]
LGTM other than comments.

File: lib/client.js (hunk 3 above)
Comment (dbkr) [inline lib/client.js]: Not really - you just call register and if it fails, it's in the response, so it's just part of the register call.

Comment (Kegsay) [inline lib/client.js]: Righto.

**Comment by Kegsay** [comment]
LGTM

**Pull request merged**